android/src/main/java/com/reactnative/googlefit/StepHistory.java

```diff
@@ -172,16 +172,35 @@ public void aggregateDataByDate(long startTime, long endTime, int bucketInterval
 
             Log.i(TAG, "  + Device    : " + device);
             if (device != null) {
+                source.putString("deviceUid", device.getUid());
                 source.putString("deviceManufacturer", device.getManufacturer());
                 source.putString("deviceModel", device.getModel());
                 switch(device.getType()) {
+                    case Device.TYPE_PHONE:  
+                        source.putString("deviceType", "phone");
+                    break;
+                    case Device.TYPE_WATCH: 
+                        source.putString("deviceType", "watch");
+                    break;
+                    case Device.TYPE_TABLET: 
+                        source.putString("deviceType", "tablet");
+                    break;
                     case Device.TYPE_CHEST_STRAP:
-                        source.putString("deviceType", "chestStrap"); break;
+                        source.putString("deviceType", "chest-strap");
+                    break;
+                    case Device.TYPE_HEAD_MOUNTED:
+                        source.putString("deviceType", "head-mounted");
+                    break;
+                    case Device.TYPE_SCALE:
+                        source.putString("deviceType", "scale");
+                    break;
+                    case Device.TYPE_UNKNOWN:
+                        source.putString("deviceType", "unknown");
+                    break;
+                    default: 
+                        source.putString("deviceType", "unknown");
+                    break;
                 }
-            } else {
-                source.putNull("deviceManufacturer");
-                source.putNull("deviceModel");
-                source.putNull("deviceType");
```

Comment:
In my opinion it's better if the fields are just missing from the response instead of setting them to `null`. Let me know if that's not a good idea.

Reply:
@aboveyunhai 

I guess that there is no correct answer of should the values be omitted or just set to `null` when they're missing, it's something that should probably be just decided and kept consistent in the codebase. 

What I can see from the current Typescript types of this library is that none of the properties are indicated to be able to be `null` but some are marked as optional using the `?` after the property name. If you think that we should stick with `null`, then there should be a refactor to the code to add `null` values to the responses where they're missing.

The fact that the native module is written in Java should in my opinion not be used to determine what coding style to use. The React Native and React libraries are written in Javascript, and they are bridged to Objective-C and Java for the native modules that have their own (Obj-C/Java) styling rules.

I can just give my two cents from an API design point of view: returning `null` as values for fields can become problematic when the API grows bigger, in which case you might be returning 20 fields in the response with values as `null`, which might mean a need for some `null` checking in the code, or for example stripping out the `null` values using `Array.prototype.filter`.

But as I said, I don't think that there is an right or wrong answer when it comes to omitting the values or having them  as `null`

Reply:
since you introduced the typescript for this part. I believe it shouldn't be a problem. If people really used some problematic null check on their side. Typescript should have a fair warning for them.